### PR TITLE
Patch hdd images before offline migration

### PR DESCRIPTION
**Pull request by dgutu-suse**
Patch all hdd images involved in offline migration scenarios.
First will be called the patching step and then the migration will start.
Patching testcases should contain variable PATCH=1

### CODE REVIEW

**Comment by dgutu-suse** [comment]
You may take a look at some of them - http://ain.arch.suse.de/tests/overview?distri=sle&version=12-SP3&build=0192&groupid=4

**Comment by dzedro** [comment]
It can be done also without saving additional image, patch in test like [here](https://openqa.suse.de/tests/670120) patch instead of ssh_key_check, and of course it will do upgrade when UPGRADE=1
Every run there can be available new patches and we have to create new unique patched disk image.

**Comment by okurz** [comment]
how does this change relate to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/2202/files ?

**Comment by dgutu-suse** [comment]
@okurz Don't get what you want to hear

**Comment by okurz** [comment]
Well, to me it looks related. Isn't it redoing some of the same stuff? I guess you understand better than me what it is supposed to do and what #2202 is supposed to do because I don't understand :-)

**Comment by dgutu-suse** [comment]
@okurz

File: tests/update/patch_before_migration.pm [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+use utils;
+use registration;
+
+sub patching_sle11() {
+
+    my ($registration, $email, $regcode, $regcode_ha, $regcode_geo) = @_;
+    $email       = get_var("SCC_EMAIL");
+    $regcode     = get_var("SCC_REGCODE");
+    $regcode_ha  = get_var("SCC_REGCODE_HA");
+    $regcode_geo = get_var("SCC_REGCODE_GEO");
```
Comment (okurz) [inline tests/update/patch_before_migration.pm]: these should be `get_required_var`

File: tests/update/patch_before_migration.pm [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Patch SLE11* qcow2 images before migration (offline)
```
Comment (okurz) [inline tests/update/patch_before_migration.pm]: that's somehow wrong, you also patch sle12, right? better just delete this line and optionally add something to '# Summary: ' further below

File: tests/update/patch_before_migration.pm [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    if (get_var("SCC_REGCODE_GEO")) {
+        $registration = script_output("
+        suse_register -n -a email=$email -a regcode-sles=$regcode -a regcode-slehae=$regcode_ha -a regcode-slehaegeo=$regcode_geo", 90);
+    }
+    elsif (get_var("SCC_REGCODE_HA")) {
+        $registration = script_output("suse_register -n -a email=$email -a regcode-sles=$regcode -a regcode-slehae=$regcode_ha", 70);
+    }
+    else {
+        $registration = script_output("suse_register -n -a email=$email -a regcode-sles=$regcode", 90);
+    }
```
Comment (okurz) [inline tests/update/patch_before_migration.pm]: how about setting the options in a string first, e.g. something like

```perl
my $regopts;
if (get_required_var('SCC_REGCODE_GEO')) {
    $regopts = "regcode-sles=$regcode -a regcode-slehae=$regcode_ha -a regcode-slehaegeo=$regcode_geo";
}
elsif (get_required_var('SCC_REGCODE_HA')) {
    $regopts = "regcode-sles=$regcode -a regcode-slehae=$regcode_ha";
}
else {
    $regopts = "regcode-sles=$regcode";
}
$registration = script_output("suse_register -n -a email=$email $regopts");
```

should make it a bit more readable and prevents that line break in line 31/32

**Comment by okurz** [comment]
please rebase on current master and make sure your "os-autoinst" subfolder is up to date to use an updated style check. You may have to create a symlink to `.perltidyrc` manually, e.g. with `ln -s os-autoinst/.perltidyrc`

**Comment by dgutu-suse** [comment]
I'm closing this PR because I re-worked testcase so this one is useless.

**Comment by okurz** [comment]
@dgutu-suse please don't close PRs just because you are reworking your git history

**Pull request closed without merging**